### PR TITLE
Filter inbound requests

### PR DESCRIPTION
**Pull request by abhay**
* Only allow POST requests to pass through to underlying RPC.
* Filters Invalid JSON
* Filters empty RPC batches

### CODE REVIEW

File: src/tests/index.test.ts [hunk 1]
```diff
@@ -18,6 +18,53 @@ describe('index', () => {
...
+					Host: 'solana-rpc.web.helium.io',
+				},
+				body: '{"jsonrpc": "2.0", }',
+			}
+		) as unknown as Parameters<typeof worker.fetch>[0];
+
+		const resp = await worker.fetch(request, originalEnv);
+
+		expect(resp.status).toEqual(400);
+		console.log(resp)
```
Comment (deasydoesit) [inline src/tests/index.test.ts:62]: Needed?

File: src/index.ts [hunk 2]
```diff
@@ -61,7 +61,31 @@ export default {
...
+				statusText: 'Only POST requests are allowed',
+			});
+		}
+
 		const payload = await request.text();
+
+		try {
+			const data = JSON.parse(payload);
+
+			if (data.length == 0) {
```
[the remark this answers is not in the document]
Comment (abhay) [reply src/index.ts:76]: Reason: I'm bad at JS.

Comment (abhay) [inline src/index.ts:76]:
```suggestion
			if (data.length === 0) {
```

Comment (abhay) [inline src/index.ts:76]:
```suggestion
			if (data.length === 0) {
```

File: src/tests/index.test.ts (hunk 1 above)
Comment (abhay) [inline src/tests/index.test.ts:62]:
```suggestion
		expect(resp.status).toEqual(400);
```

Comment (abhay) [reply src/tests/index.test.ts]: Negative

**Review by deasydoesit — APPROVED** [review]

**Pull request merged**